packages/product/src/services/product-module-service.ts

```diff
@@ -1116,22 +1116,41 @@ export default class ProductModuleService<
     return await this.productService_.softDelete(productIds, sharedContext)
   }
 
-  async restore(
+  async restore<
+    TReturnableLinkableKeys extends string = Lowercase<
+      keyof typeof LinkableKeys
+    >
+  >(
     productIds: string[],
+    {
+      returnLinkableKeys,
+    }: { returnLinkableKeys?: TReturnableLinkableKeys[] } = {
+      returnLinkableKeys: [],
+    },
     sharedContext: Context = {}
-  ): Promise<ProductTypes.ProductDTO[]> {
-    const products = await this.restore_(productIds, sharedContext)
+  ): Promise<Record<Lowercase<keyof typeof LinkableKeys>, string[]> | void> {
+    const [_, cascadedEntitiesMap] = await this.restore_(
+      productIds,
+      sharedContext
+    )
 
-    return this.baseRepository_.serialize<ProductTypes.ProductDTO[]>(products, {
-      populate: true,
-    })
+    let mappedCascadedEntitiesMap
+    if (returnLinkableKeys) {
+      mappedCascadedEntitiesMap = mapObjectTo<
```

Comment:
comment: Can I get you to add a sentence or two on what is happening here? It's not immediately obvious to me 😅

Reply:
will add a brief comment to it:
```
// Map internal table/column names to their respective external linkable keys
// eg: product.id = product_id, variant.id = variant_id
```

This is required when softDeleting things, all the internal relationships that were affected will be returned so the RemoteLink can cascade delete all the relations.
For example, if you delete a product, this will return the id of the product deleted, all the ids of variants related to it, product options, etc...
That will create the cascade effect to delete everything that is linked to any of these things.

Reply:
Thanks!